packages/jest-core/src/cli/index.ts

```diff
@@ -167,13 +155,28 @@ const buildContextsAndHasteMaps = async (
   return {contexts, hasteMapInstances};
 };
 
-const _run10000 = async (
+/**
+ * Runs Jest either in watch mode or as a one-off. This is a lower-level API than `runCLI` and is intended for internal use by `runCLI` or externally.
+ * Note that `process.exit` might be called when using `globalConfig.watch` or `globalConfig.watchAll` is true.
+ *
+ * @param globalConfig The global configuration to use for this run. It can be obtained using `readConfigs` (imported from 'jest-config').
+ * @param configs The project configurations to run. It can be obtained using `readConfigs` (imported from 'jest-config').
+ * @param warnForDeprecations Whether or not to warn for deprecation messages when `globalConfig.watch` or `globalConfig.watchAll` is true.
```

Comment:
By the way, `readConfigs()` returns `hasDeprecationWarnings` as well. Or I misunderstood? Hm.. Not sure why it is renamed here.

Reply:
I've renamed it to more clearly communicate intent. I have noticed that it is only used to warn the user about deprecation warnings. 

If we create a separate run, similar to `runCLI` than I suggest we can change it back and simply not expose this on `runCore`.